### PR TITLE
TASK: Remove shallow `ContentGraphProjection` decorator

### PR DESCRIPTION
**Pull request by mhsdesign**
**Upgrade instructions**

**Review instructions**

The `ContentGraphProjection` didnt add anything but complexity. It was only used by the doctrine dbal impl but not from the postgres client because it turns out the concept is overhauled. Its just a dead decorator with no apparent use.

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.ContentGraph.DoctrineDbalAdapter/src/DoctrineDbalContentGraphProjection.php [hunk 1]
```diff
@@ -150,6 +151,11 @@ public function reset(): void
         $this->getState()->forgetInstances();
     }
 
+    public function markStale(): void
+    {
+        $this->getState()->forgetInstances();
+    }
```
Comment (bwaidelich) [inline Neos.ContentGraph.DoctrineDbalAdapter/src/DoctrineDbalContentGraphProjection.php:155]: I don't understand why this can be removed?!

**Review by bwaidelich — COMMENTED** [review]
Makes sense to me, thanks.
Just one question re `WithMarkStaleInterface"

File: Neos.ContentGraph.DoctrineDbalAdapter/src/DoctrineDbalContentGraphProjection.php (hunk 1 above)
Comment (mhsdesign) [reply Neos.ContentGraph.DoctrineDbalAdapter/src/DoctrineDbalContentGraphProjection.php:155]: not removed. It has been copied from the decorator `ContentGraphProjection` back to the `DoctrineDbalContentGraphProjection`

(basically reverting what we tree did when we watched over christians shoulder in the the content graph thing)

Comment (mhsdesign) [reply Neos.ContentGraph.DoctrineDbalAdapter/src/DoctrineDbalContentGraphProjection.php:155]: behaviour is 100% the same (test fails without this addition btw)

Comment (bwaidelich) [reply Neos.ContentGraph.DoctrineDbalAdapter/src/DoctrineDbalContentGraphProjection.php:155]: oh man.. I read the diff backwards.. somehow.. Time for weekend :)

**Review by bwaidelich — APPROVED** [review]

**Comment by mhsdesign** [comment]
I guess this can be merged before stale again // cc @kitsunet ?

**Review by kitsunet — APPROVED** [review]
Indeed

**Pull request merged**